src/ipfs.rs

```diff
@@ -0,0 +1,29 @@
+use std::{collections::HashMap, path::Path};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use serde_json::Value;
+
+pub mod daemon;
+
+#[cfg(test)]
+pub mod tests;
+
+#[async_trait]
+pub trait Ipfs {
+    /// This method uploads a file or directory at a given path to the IPFS swarm you are
+    /// currently connected to.
+    async fn add(&self, path:&Path) -> Result<HashMap<String, String>>;
+    /// This method connects to the given address, adding the address to the current swarm
+    async fn connect_to(&self, peer_id:&str) -> Result<()>;
+    /// This method returns a list of all the addresses that are currently connected
+    async fn get_connected(&self) -> Result<Vec<String>>;
+    /// This method changes the value of a given property in the IPFS config
+    /// 
+    /// Ex. `ipfs.set_config("Datastore.StorageMax", "11GB")`
+    async fn set_config(&self, property:&str, val:&Value) -> Result<()>;
+    /// This method returns the value a property in the IPFS config
+    /// 
+    /// Ex. `ipfs.get_config("Datastore.StorageMax")`
```

Comment:
So, doc tests will fail if I attempt to do them properly because it gets a connection refused error. I can either remove those examples completely, find a way to do the doc tests without having to run them when cargo test is run, or leave them how they are.

Reply:
doctests can be made ignore run with `ignore` keyword after the three backticks that assume "run rust"
If you want to gate the doctest to make the `ignore` based on e.g. `cfg()` you can use `cfg_attr` like this:
https://github.com/dalek-cryptography/ed25519-dalek/blob/release/2.0/src/lib.rs#L21-22

Reply:
Changed it to be an ignored doc test